### PR TITLE
Properties refactoring

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR moves various property-related methods out of Communicator into a new static "extension" class, Properties.

### CODE REVIEW

**Review by externl — APPROVED** [review]

File: csharp/src/Ice/Properties.cs [hunk 1]
```diff
@@ -0,0 +1,609 @@
...
+
+            foreach (KeyValuePair<string, string> p in parsedArgs)
+            {
+                into[p.Key] = p.Value;
+            }
+
+            args = remaining.ToArray();
+        }
+
+        internal static void CheckForUnknownProperties(this Communicator communicator, string prefix)
```
Comment (pepone) [inline csharp/src/Ice/Properties.cs:310]: Wouldn't make more sense to have this as a private static on the Reference?

**Review by pepone — APPROVED** [review]
Looks good!

File: csharp/src/Ice/Properties.cs [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+                        ex);
+                }
+            }
+            return null;
+        }
+
+        /// <summary>Loads an Ice configuration file.</summary>
+        /// <param name="into">The property dictionary into which the loaded properties are added.</param>
+        /// <param name="configFile">The path to the Ice configuration file to load.</param>
+        public static void LoadIceConfigFile(this Dictionary<string, string> into, string configFile)
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply csharp/src/Ice/Properties.cs:235]: Parse are typically static methods, like `int.Parse(s)`. Here it's an extension method used as `props.LoadIceConfigFile(myFile)`. I find LoadIceConfigFile more natural here.

File: csharp/src/Ice/Properties.cs (hunk 1 above)
Comment (bernardnormier) [reply csharp/src/Ice/Properties.cs]: Yes, fixed.

**Pull request merged**